search-impl/src/test/java/com/example/core/TopicStub.java

```diff
@@ -49,13 +54,21 @@ public TopicStub(Materializer materializer) {
                         .via(flow)
                         .toMat(Sink.ignore(), Keep.both())
                         .run(materializer);
-                sourceActor = pair.first();
+                sourceActor.complete(pair.first());
                 return pair.second();
             }
         };
     }
 
-    public Supplier<ActorRef> actorSupplier() {
-        return () -> sourceActor;
+    /**
+     * @return a {@link ProducerStub} that clients (test code) can use to send messages through this {@link TopicStub}.
+     */
+    public ProducerStub<T> producerStub() {
+        return (T message) ->
+                sourceActor.thenApply(topicActor -> {
```

Comment:
If you backed `TopicStub` with an actor that was created when it was created, then this method could be replaced with a simple `send` method that would send directly to the actor.